src/main/java/hudson/plugins/git/extensions/impl/PathRestriction.java

```diff
@@ -127,8 +127,12 @@ public Boolean isRevExcluded(GitSCM scm, GitClient git, GitChangeSet commit, Tas
             }
         }
 
-        // If every affected path is excluded, return true.
-        if (includedPaths.size() == excludedPaths.size()) {
+       if (excluded.isEmpty() && !included.isEmpty() && includedPaths.isEmpty()) {
```

Comment:
Your timing is perfect.  I had just completed the merge when you detected that.  I'll fix it with a subsequent commit.